File: app/src/main/java/com/yhjoo/dochef/utils/MyAppGlideModule.java
package com.yhjoo.dochef.utils;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;

import com.bumptech.glide.Glide;
import com.bumptech.glide.GlideBuilder;
import com.bumptech.glide.Registry;
import com.bumptech.glide.annotation.GlideModule;
import com.bumptech.glide.module.AppGlideModule;
import com.firebase.ui.storage.images.FirebaseImageLoader;
import com.google.firebase.storage.StorageReference;

import java.io.InputStream;

@GlideModule
public final class MyAppGlideModule extends AppGlideModule {

    @Override
    public void registerComponents(Context context, Glide glide, Registry registry) {
        registry.append(StorageReference.class, InputStream.class, new FirebaseImageLoader.Factory());
    }

    @Override
    public void applyOptions(@NonNull Context context, @NonNull GlideBuilder builder) {
        super.applyOptions(context, builder);
//        builder.setLogLevel(Log.ERROR);
    }
}

File: app/src/main/java/com/yhjoo/dochef/utils/MyAppGlideModule.kt
package com.yhjoo.dochef.utils

import com.bumptech.glide.annotation.GlideModule
import com.bumptech.glide.module.AppGlideModule

@GlideModule
class MyAppGlideModule : AppGlideModule()
//{
//    // Todo
////  FirebaseImageLoader 이런게 없음
//    override fun registerComponents(context: Context, glide: Glide, registry: Registry) {
////        registry.append(
////            StorageReference::class.java,
////            InputStream::class.java,
////            FirebaseImageLoader.Factory()
////        )
//    }
//
//    override fun applyOptions(context: Context, builder: GlideBuilder) {
//        super.applyOptions(context, builder)
////        builder.setLogLevel(Log.ERROR);
//    }
//}